### PR TITLE
Serialize api activation via a property

### PR DESCRIPTION
**Pull request by jaivasanth-google**
…roperty

### CODE REVIEW

File: examples/v2/project_creation/apis.py [hunk 1]
```diff
@@ -19,14 +19,19 @@ def GenerateConfig(context):
...
   billing = context.properties['billing']
+  concurrent_api_activation = context.properties['concurrent_api_activation']
 
   resources = []
-  for api in context.properties['apis']:
+  for index, api in enumerate(context.properties['apis']):
+    dependsOn = [project_id, billing]
+    # Serialize the activation of all the apis by making api_n depend on api_n-1
+    if (not concurrent_api_activation) and index != 0:
+      dependsOn.append(project_id + '-' + context.properties['apis'][index-1])
```
Comment (katyp) [inline examples/v2/project_creation/apis.py]: variable name should be depends-on

Comment (katyp) [inline examples/v2/project_creation/apis.py]: Any use in standardizing the API name now that it's created in two different places? Like, write a function that can create a well-formatted name using this `project_id + '-' + api_name` pattern?

File: examples/v2/project_creation/apis.py [hunk 2]
```diff
@@ -19,14 +19,19 @@ def GenerateConfig(context):
 
   project_id = context.properties['project']
   billing = context.properties['billing']
+  concurrent_api_activation = context.properties['concurrent_api_activation']
 
   resources = []
-  for api in context.properties['apis']:
+  for index, api in enumerate(context.properties['apis']):
+    dependsOn = [project_id, billing]
+    # Serialize the activation of all the apis by making api_n depend on api_n-1
```
Comment (katyp) [inline examples/v2/project_creation/apis.py]: Do you think it would be useful to still do a few api's concurrently? Like 4 at a time, or so? Could be significantly faster, and might not exceed quota?

Comment (jaivasanth-google) [inline examples/v2/project_creation/apis.py]: I thought about did not know where to draw the line. I could perhaps change it  so that the concurrency level is controlled by a parameter and default it to 1 or 2. That would be a pretty involved change and will require unit tests, and prefer to do that in a separate CL.

**Comment by jaivasanth-google** [comment]
Feedback addressed, PTAL.

File: examples/v2/project_creation/project.py [hunk 3]
```diff
@@ -50,7 +50,8 @@ def GenerateConfig(context):
       'properties': {
           'project': project_id,
           'billing': billing_name,
-          'apis': context.properties['apis']
+          'apis': context.properties['apis'],
+          'concurrent_api_activation': context.properties['concurrent_api_activation']
```
Comment (katyp) [inline examples/v2/project_creation/project.py]: Blergh. Sorry, Jai. This line is too long. Linter says it needs to be <80 and it's at 86. How strict do we want to be with the linting on this repo?

File: examples/v2/project_creation/apis.py [hunk 4]
```diff
@@ -19,14 +19,19 @@ def GenerateConfig(context):
...
   billing = context.properties['billing']
+  concurrent_api_activation = context.properties['concurrent_api_activation']
 
   resources = []
-  for api in context.properties['apis']:
+  for index, api in enumerate(context.properties['apis']):
+    depends_on = [project_id, billing]
+    # Serialize the activation of all the apis by making api_n depend on api_n-1
+    if (not concurrent_api_activation) and index != 0:
+      depends_on.append(ApiResourceName(project_id, context.properties['apis'][index-1]))
```
Comment (katyp) [inline examples/v2/project_creation/apis.py]: "Line too long (86/80) [line-too-long]"

File: examples/v2/project_creation/apis.py [hunk 5]
```diff
@@ -35,3 +40,6 @@ def GenerateConfig(context):
     })
 
   return {'resources': resources}
+
```
Comment (katyp) [inline examples/v2/project_creation/apis.py]: "Need 2 blank lines before top level def [g-wrong-blank-lines]"

**Comment by katyp** [comment]
I'll look into getting a linter to run on Github. I don't think Google's internal python style matches public python anyway, so it might be good for us to have a different linter in this repo than `g4 lint` anyway.
